core/domain/exp_jobs_one_off_test.py

```diff
@@ -92,6 +92,240 @@ def run_job_for_deleted_exp(
+    and skin_customizations.
+    """
+
+    skill_tags = (
+        datastore_services.StringProperty(indexed=True, repeated=True))
+    default_skin = (
+        datastore_services.StringProperty(default='conversation_v1'))
+    skin_customizations = datastore_services.JsonProperty(indexed=False)
+
+    def _trusted_commit(
+            self, committer_id, commit_type, commit_message, commit_cmds):
+        """Override this to escape recursion, which will otherwise occur
+        when super() is used in a mocked class.
+        """
+
+        base_models.VersionedModel._trusted_commit( # pylint: disable=protected-access
+            self, committer_id, commit_type, commit_message, commit_cmds)
+
+
+class RemoveDeprecatedExplorationModelFieldsOneOffJobTests(
+        test_utils.GenericTestBase):
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            exp_jobs_one_off.RemoveDeprecatedExplorationModelFieldsOneOffJob
+            .create_new()
+        )
+        (
+            exp_jobs_one_off.RemoveDeprecatedExplorationModelFieldsOneOffJob
+            .enqueue(job_id))
+        self.assertEqual(
+            self.count_jobs_in_mapreduce_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            exp_jobs_one_off.RemoveDeprecatedExplorationModelFieldsOneOffJob
+            .get_output(job_id))
+        eval_output = [ast.literal_eval(stringified_item) for
+                       stringified_item in stringified_output]
+        return eval_output
+
+    def test_one_exp_models_with_deprecated_field_skill_tag(self):
+        with self.swap(
+            exp_models,
+            'ExplorationModel',
+            MockExplorationModelWithDeprecatedFields
+        ):
+            exp_model1 = exp_models.ExplorationModel(
+                id='exp_id1',
+                category='category',
+                title='title',
+                objective='objective',
+                language_code='en',
+                tags=[],
+                blurb='',
+                author_notes='',
+                init_state_name=feconf.DEFAULT_INIT_STATE_NAME,
+                param_specs={},
+                param_changes=[],
+                skill_tags=['tag1', 'tag2'],
+            )
+            rights_manager.create_new_exploration_rights('exp_id1', 'uid_1')
+            commit_message = 'New exploration created with title \'title\'.'
+            exp_model1.commit('uid_1', commit_message, [{
+                'cmd': 'create_new',
+                'title': 'title',
+                'category': 'category',
+            }])
+
+            self.assertIsNotNone(exp_model1.skill_tags)
+
+            self.assertIn('skill_tags', exp_model1._values) # pylint: disable=protected-access
+            self.assertIn('skill_tags', exp_model1._properties) # pylint: disable=protected-access
```

Comment:
This needs to be removed since we discovered that sometimes the value is not in the `_properties`, ditto in other places where we have the same problem.

Reply:
The value is in `_properties` everytime when it is in `_values`. This was resolved by merging 3 tests into 1 test.